bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -646,9 +656,18 @@ public ServerConfiguration setJournalDirsName(String[] journalDirs) {
      *
      * @return ledger dir names, if not provided return null
      */
+    public String[] getLedgerDirWithoutDefault() {
+        return this.getStringArray(LEDGER_DIRS);
+    }
+
+    /**
+     * Get dir names to store ledger data
+     *
+     * @return ledger dir names, if not provided return null
+     */
     public String[] getLedgerDirNames() {
         String[] ledgerDirs = this.getStringArray(LEDGER_DIRS);
-        if (null == ledgerDirs) {
+        if ((null == ledgerDirs) || (0 == ledgerDirs.length)) {
```

Comment:
For a bookie joining first time, there is no cookie. So cookie doesn't prevent this. 

but the behavior is same as not configuring ledger dirs. I am fine changing the behavior to be more strict but this would have to be a separate jira.